### ISSUE TITLE
Recursively check for outdated dependencies

### ISSUE THREAD

**Issue by lorenzomigliorero** [changesets/changesets#1024]
## Affected Packages

**@changesets/apply-release-plan**

## Problem

Our monorepo adopts this dependencies tree:

```mermaid
graph TD;
Package-A-->Package-B;
Package-A-->Package-C;
Package-B-->Package-D;
```

Updating `Package-A` successfully causes a patch update of `Package-B` and `Package-C`.
Instead,  `Package-D` remains out-of-date, cause it's not a direct dependency of `Package-A`.

## Proposed solution

Recursively check for outdated dependencies inside the `versionPackage` method. https://github.com/changesets/changesets/blob/c1401716cf5ee839aaa02ea7ff8f23f8af8bf5b0/packages/apply-release-plan/src/version-package.ts#L17

**Comment by Andarist** [comment]
Could you list your `package.json` content? Package versions and dependency ranges matter for this. It would be great if you could provide a repro case that I could investigate

**Comment by Andarist** [comment]
Right, `versionPackages` might not propagate that change recursively but IIRC the `assembleReleasePlan` should do that. I'll have to investigate this! Thanks for the report
